ext/fast_jsonparser/fast_jsonparser.cpp

```diff
@@ -93,36 +91,44 @@ static VALUE rb_fast_jsonparser_load(VALUE self, VALUE arg)
     return Qnil;
 }
 
-static VALUE rb_fast_jsonparser_load_many(VALUE self, VALUE arg)
+static VALUE rb_fast_jsonparser_load_many(VALUE self, VALUE arg, VALUE batch_size)
 {
     Check_Type(arg, T_STRING);
+    Check_Type(batch_size, T_FIXNUM);
 
-    dom::parser parser;
-    auto [docs, error] = parser.load_many(RSTRING_PTR(arg));
-    if (error == SUCCESS)
-    {
-        for (dom::element doc : docs)
+    try {
+        dom::parser parser;
+        auto [docs, error] = parser.load_many(RSTRING_PTR(arg), FIX2INT(batch_size));
+        if (error == SUCCESS)
         {
-            if (rb_block_given_p())
```

Comment:
This `rb_block_given_p()` is kinda wasteful. If you don't pass a block, you might as well raise anyway.